src/ucp/core/ucp_worker.c

```diff
@@ -1656,16 +1710,28 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
         goto err_close_ifaces;
     }
 
+    /* Open all resources as connection managers on this worker */
+    status = ucp_worker_add_resource_cms(worker);
+    if ((status == UCS_ERR_UNSUPPORTED) &&
```

Comment:
maybe check num_cms rather than return value?
if this function returns error code it should fail regardless of proto